### ISSUE TITLE
used_underscore_binding trigged on #[derive(Serialize)] using serde

### ISSUE THREAD

**Issue by compressed** [rust-lang/rust-clippy#852]
Similar to https://github.com/Manishearth/rust-clippy/issues/536, I'm getting this clippy warning on the expansion of `Serialize`.

`Cargo.toml`

``` toml
[dependencies]
clippy = "0.0.63"
serde = "0.7.0"
serde_macros = "0.7.2"
```

`lib.rs`

``` rust
#![feature(custom_derive, plugin)]
#![plugin(serde_macros, clippy)]

extern crate serde;

#[derive(Serialize)]
pub struct A {
    a: String,
}
```

Returns:

```
src/lib.rs:7:10: 7:19 warning: used binding which is prefixed with an underscore. A leading underscore signals that a binding will not be used., #[warn(used_underscore_binding)] on by default
src/lib.rs:7 #[derive(Serialize)]
                      ^~~~~~~~~
src/lib.rs:7:10: 7:19 note: in this expansion of try! (defined in <std macros>)
src/lib.rs:7:10: 7:19 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#used_underscore_binding
```

**Comment by llogiq** [comment]
Good catch! It seems this lint should have a stricter macro check.

**Comment by euclio** [comment]
FWIW, `maplit`'s `btreemap!` and `map!` macros also trigger this warning.

**Comment by llogiq** [comment]
Also `bitflags!`.

**Comment by indiv0** [comment]
Any progress on this?
